sdk/iOS/test/MSSyncTableTests.m

```diff
@@ -1082,9 +1083,348 @@ -(void) testPushNetworkTimeoutWithErroredOps
+        
+        return data;
+    };
+    
+    // Perform an insert
+    XCTestExpectation *insertExpectation = [self expectationWithDescription:@"insert expectation"];
+    [todoTable insert:@{ @"id" : @"id1", @"text" : @"client" } completion:^(NSDictionary *item, NSError *error) {
+        [insertExpectation fulfill];
+    }];
+    
+    [self waitForExpectationsWithTimeout:10 handler:nil];
+    
+    // Perform a push
+    pushOperation = [filteredClient.syncContext pushWithCompletion:^(NSError *error) {
+        
+        // Cancellation point 2
+        if (requestedCancellationPoint == 2)
+            [pushOperation cancel];
+    }];
+    
+    // Cancellation point 3
+    if (requestedCancellationPoint == 3)
+        [pushOperation cancel];
+
+    [self verifyFinished:pushOperation];
+
+    int synchronizedItemCount = [self synchronizedItemCount];
+
+    // Perform another insert
+    insertExpectation = [self expectationWithDescription:@"insert expectation"];
+    [todoTable insert:@{ @"id" : @"id2", @"text" : @"client" } completion:^(NSDictionary *item, NSError *error) {
+        [insertExpectation fulfill];
+    }];
+    
+    [self waitForExpectationsWithTimeout:10 handler:nil];
+
+    XCTAssertEqual([self performVanillaPush], 2, "Expected all inserted items to have been pushed to the server by now");
+
+    return synchronizedItemCount;
+}
+
+// Performs a push operation and returns the count of all client items pushed so far
+-(int) performVanillaPush
+{
+    // Define a filter that returns an appropriate server response for the requested item
+    MSTestFilter *filter = [MSTestFilter new];
+    MSClient *filteredClient = [client clientWithFilter:filter];
+
+    filter.ignoreNextFilter = YES;
+    filter.onInspectResponseData = ^(NSURLRequest *request, NSData *data) {
+        NSError *error = nil;
+        
+        NSDictionary *item = [filteredClient.serializer itemFromData:request.HTTPBody withOriginalItem:nil ensureDictionary:YES orError:&error];
+        XCTAssertNil(error);
+        
+        NSString *responseString = [NSString stringWithFormat:@"{\"id\": \"%@\", \"text\":\"server\"}", item[@"id"]];
+        NSData *pushResponse = [responseString dataUsingEncoding:NSUTF8StringEncoding];
+
+        return pushResponse;
+    };
+
+    // Perform push
+    XCTestExpectation *pushExpectation = [self expectationWithDescription:@"push expectation"];
+    [filteredClient.syncContext pushWithCompletion:^(NSError *error) {
+        XCTAssertNil(error, "Expected successful push");
+        [pushExpectation fulfill];
+    }];
+    [self waitForExpectationsWithTimeout:10 handler:nil];
+    
+    return [self synchronizedItemCount];
+}
+
+// Returns the number of synchronized items, i.e. items that are present on the server as well as the client.
+// An item is synchronized if it was successfully pushed to the server from the client OR
+// if it was successfully pulled from the server to the client.
+-(int) synchronizedItemCount
```

Comment:
Do you mean the upsertedItems, upsertedCalls, etc metrics? They track the upserted items in all tables (the table we are inserting in, operations table, errors table) not just the table we are inserting in. We just care about the final state of the user table.